### PR TITLE
Suppress ranges

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/bessel/regular/impl/cyl_bessel_j0.hpp [hunk 1]
```diff
@@ -27,34 +23,35 @@ cyl_bessel_j0_(EVE_SUPPORTS(cpu_), T a0) noexcept
...
-                                               -1.2441026745835638459e+00,
-                                               -8.8033303048680751817e-03};
-      constexpr std::array<elt_t, 6> QS     = {5.7105024128512061905e+03,
-                                               1.1951131543434613647e+04,
-                                               7.2642780169211018836e+03,
-                                               1.4887231232283756582e+03,
-                                               9.0593769594993125859e+01,
-                                               1.0};
+      using A6 = kumi::result::generate_t<6, elt_t>;
+      static const A6 PC     = {2.2779090197304684302e+04,
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_j0.hpp:27]: should be constexpr ?

File: include/eve/module/bessel/regular/impl/cyl_bessel_i1.hpp [hunk 2]
```diff
@@ -27,14 +24,15 @@ cyl_bessel_i1_(EVE_SUPPORTS(cpu_), T x) noexcept
...
-        constexpr std::array<elt_t, 8> P    = {8.333333221e-02f,
-                                               6.944453712e-03f,
-                                               3.472097211e-04f,
-                                               1.158047174e-05f,
-                                               2.739745142e-07f,
-                                               5.135884609e-09f,
-                                               5.262251502e-11f,
-                                               1.331933703e-12f};
+        using A8= kumi::result::generate_t<8, elt_t>;
+        static const A8 P    = {8.333333221e-02f,
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_i1.hpp:28]: should be constexpr

File: include/eve/module/bessel/regular/impl/cyl_bessel_i0.hpp [hunk 3]
```diff
@@ -27,54 +25,58 @@ cyl_bessel_i0_(EVE_SUPPORTS(cpu_), T x) noexcept
...
-                                            1.73560257755821695e-03f,
-                                            6.96166518788906424e-05f,
-                                            1.89645733877137904e-06f,
-                                            4.29455004657565361e-08f,
-                                            3.90565476357034480e-10f,
-                                            1.48095934745267240e-11f};
-        T                              a = sqr(x / 2);
-        return inc(a * reverse_horner(a, P));
+      using A9 = kumi::result::generate_t<9, elt_t>;
+      static const A9 P = {1.00000003928615375e+00f,
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_i0.hpp:29]: should be constexpr

File: include/eve/module/bessel/regular/impl/cyl_bessel_j1.hpp [hunk 4]
```diff
@@ -24,38 +20,39 @@ cyl_bessel_j1_(EVE_SUPPORTS(cpu_), T x) noexcept
...
-                                             0.0};
-    constexpr std::array<elt_t, 7> QS     = {7.0871281941028743574e+05,
-                                             1.8194580422439972989e+06,
-                                             1.4194606696037208929e+06,
-                                             4.0029443582266975117e+05,
-                                             3.7890229745772202641e+04,
-                                             8.6383677696049909675e+02,
-                                             1.0};
+    using A7 = kumi::result::generate_t<7, elt_t>;
+    A7 PC     = {-4.4357578167941278571e+06,
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_j1.hpp:24]: should be constexpr

File: include/eve/module/bessel/regular/impl/cyl_bessel_k0.hpp [hunk 5]
```diff
@@ -28,13 +25,19 @@ cyl_bessel_k0_(EVE_SUPPORTS(cpu_), T x) noexcept
       T y(1.137250900268554688);
       if constexpr( std::same_as<elt_t, float> )
       {
-        constexpr std::array<elt_t, 3> P = {
-            -1.372508979104259711e-01f, 2.622545986273687617e-01f, 5.047103728247919836e-03f};
-        constexpr std::array<elt_t, 3> Q = {
-            1.000000000000000000e+00f, -8.928694018000029415e-02f, 2.985980684180969241e-03f};
+        using A3 = kumi::result::generate_t<3, elt_t>;
+        static const A3 P = {
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_k0.hpp:29]: should be constexpr

File: include/eve/module/bessel/regular/impl/cyl_bessel_k1.hpp [hunk 6]
```diff
@@ -28,38 +25,59 @@ cyl_bessel_k1_(EVE_SUPPORTS(cpu_), T x) noexcept
       if constexpr( std::same_as<elt_t, float> )
       {
         T                              y(8.695471287e-02f);
-        constexpr std::array<elt_t, 3> P = {-3.621379531e-03f, 7.131781976e-03f, -1.535278300e-05f};
-        constexpr std::array<elt_t, 3> Q = {1.000000000e+00f, -5.173102701e-02f, 9.203530671e-04f};
+        using A3 = kumi::result::generate_t<3, elt_t>;
+        static const A3 P = {
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_k1.hpp:29]: should be constexpr

File: include/eve/module/bessel/regular/impl/cyl_bessel_y0.hpp [hunk 7]
```diff
@@ -34,102 +32,114 @@ cyl_bessel_y0_(EVE_SUPPORTS(cpu_), T x) noexcept
...
-                                       1.0102532948020907590e+04,
-                                       -1.8402381979244993524e+01};
-    constexpr std::array<v_t, 6> Q1 = {5.8873865738997033405e+11,
-                                       8.1617187777290363573e+09,
-                                       5.5662956624278251596e+07,
-                                       2.3889393209447253406e+05,
-                                       6.6475986689240190091e+02,
-                                       1.0};
+    using A6 = kumi::result::generate_t<6, elt_t>;
+    static const A6 P1 = {1.0723538782003176831e+11,
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_y0.hpp:44]: should be constexpr

File: include/eve/module/bessel/regular/impl/cyl_bessel_y1.hpp [hunk 8]
```diff
@@ -35,27 +33,28 @@ cyl_bessel_y1_(EVE_SUPPORTS(cpu_), T x) noexcept
...
-                                       -3.1714424660046133456e+02};
-    constexpr std::array<v_t, 7> Q1 = {3.0737873921079286084e+14,
-                                       4.1272286200406461981e+12,
-                                       2.7800352738690585613e+10,
-                                       1.2250435122182963220e+08,
-                                       3.8136470753052572164e+05,
-                                       8.2079908168393867438e+02,
-                                       1.0};
+    using A7 = kumi::result::generate_t<7, elt_t>;
+    static const A7 P1 = {4.0535726612579544093e+13,
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/cyl_bessel_y1.hpp:44]: should be constexpr

File: include/eve/module/polynomial/detail/horner_impl.hpp [hunk 9]
```diff
@@ -46,54 +46,54 @@ horner_impl(D const& d, T0 const& xx, Cs... cs) noexcept
...
-  {
-    using std::advance;
-    auto dfma = d(fma);
-    auto that = r_t(*cur);
-    auto step = [&](auto that, auto arg) { return dfma(x, that, arg); };
-    for( advance(cur, 1); cur != last; advance(cur, 1) ) that = step(that, *cur);
-    return that;
-  }
-}
+// template<decorator D, value T0, range R>
```
Comment (jfalcou) [inline include/eve/module/polynomial/detail/horner_impl.hpp:49]: delete don't  comment

File: include/eve/module/polynomial/detail/horner_impl.hpp [hunk 10]
```diff
@@ -46,54 +46,54 @@ horner_impl(D const& d, T0 const& xx, Cs... cs) noexcept
...
-      auto [pi, epi] = two_prod(x, that);
-      auto [th, si]  = two_add(pi, arg);
-      that           = th;
-      err            = fma(err, x, epi + si);
-    };
-    for( advance(cur, 1); cur != last; advance(cur, 1) ) step(r_t(*cur));
-    return that + err;
-  }
-}
+// template<value T0, range R>
```
Comment (jfalcou) [inline include/eve/module/polynomial/detail/horner_impl.hpp:71]: delete dont comment

File: include/eve/module/special/regular/impl/lambert.hpp [hunk 11]
```diff
@@ -21,14 +20,15 @@ EVE_FORCEINLINE constexpr auto
 lambert_serie_utility(T r) noexcept
 {
   using elt_t = element_type_t<T>;
+  using A3 = kumi::result::generate_t<3, elt_t>;
   if constexpr( sizeof(elt_t) == 8 )
   {
-    constexpr std::array<elt_t, 3> P = {
+    A3 P = {
```
Comment (jfalcou) [inline include/eve/module/special/regular/impl/lambert.hpp:26]: should be constexpr

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/bessel/regular/impl/cyl_bessel_j1.hpp (hunk 4 above)
Comment (jfalcou) [reply include/eve/module/bessel/regular/impl/cyl_bessel_j1.hpp:24]: This one is still not constexpr

Comment (jtlap) [reply include/eve/module/bessel/regular/impl/cyl_bessel_j1.hpp]: je crois que j'ai tout fait là

**Review by jfalcou — APPROVED** [review]

**Pull request merged**